external_parser/event_processors/joined_event.h

```diff
@@ -245,24 +259,26 @@ struct ccb_joined_event : public typed_joined_event {
       }
     }
 
-    for (auto &slot : outcomes_map) {
-      if (slot.first == -1) {
-        VW::io::logger::log_warn(
-            "CCB outcome event for event: [{}] "
-            "has slot index and slot id missing. This is an activation which "
-            "is not currently supported so it will be ignored.",
-            metadata_info.event_id);
-        continue;
-      }
+    for (size_t i = 0; i < interaction_data.size(); i++) {
+      float reward = default_reward;
+      float original_reward = default_reward;
 
-      float reward = reward_function(slot.second);
-      set_cost(examples, reward, slot.first);
-    }
+      auto iterator = outcomes_map.find(i);
```

Comment:
nit: since the iterator is not used anywhere else you could collapse lines 266 and 267 into one:

`if (outcomes_map.find(i) != outcomes_map.end())`